rootfs/etc/nginx/lua/monitor.lua

```diff
@@ -27,12 +27,43 @@ local function send(payload)
   assert(s:close())
 end
 
+-- when the traffic was send to canary service, the canary service name is more
+-- appropriate for metrics.
+local function get_service_name(namespace, serviceName, alternativeUpstreamName)
```

Comment:
Why do you need to look at `namespace` and `serviceName`? Why is `alternativeUpstreamName` not enough to extract canary service name?

Reply:
The `alternativeUpstreamName` may be empty, so that will return the origin `serviceName` in this situation. And we need  `namespace` to split `alternativeUpstreamName` because of it is formated by `fmt.Sprintf("%s-%s-%v", f.Namespace, serviceName, servicePort)`.